ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,338 @@
+        else:
+            id_, ontologySource_ = self.namespaceSplit(location['id'])
+            name = location['id']
+
+        f = protocol.Feature()
+        f.featureType = protocol.OntologyTerm.fromJsonDict({
+            "name": name,
+            "id": id_,
+            "ontologySource": ontologySource_})
+        f.id = annotation['id']
+        f.featureSetId = ''
+        f.parentIds = []
+        f.attributes = protocol.Attributes.fromJsonDict({"vals": {}})
+
+        # # debugger example how to validate and capture validation errors
+        # if not protocol.Feature.validate(f.toJsonDict()):
+        #     e = exceptions.RequestValidationFailureException(
+        #         f.toJsonDict(),protocol.Feature)
+        #     print(e.message)
+        #     from IPython.core.debugger import Pdb ;        Pdb().set_trace()
+
+        id_, ontologySource_ = self.namespaceSplit(
+                                       annotation[hasObject]['val'])
+
+        fpa = protocol.FeaturePhenotypeAssociation()
+        fpa.id = annotation['id']
+        fpa.features = [f]
+        fpa.description = None
+        fpa.evidence = []
+        fpa.environmentalContexts = []
+
+        phenotypeInstance = protocol.PhenotypeInstance()
+        phenotypeInstance.type = protocol.OntologyTerm.fromJsonDict({
+            "name": annotation[hasObject]['label'],
+            "id": id_,
+            "ontologySource": ontologySource_})
+        fpa.phenotype = phenotypeInstance
+
+        #  ECO or OBI is recommended
+        if source in annotation:
+            if not isinstance(annotation[source], list):
+                annotation[source] = [annotation[source]]
+            for src in annotation[source]:
+                evidence = protocol.Evidence()
+                evidence.evidenceType = protocol.OntologyTerm()
+                id_, ontologySource_ = self.namespaceSplit(src['val'])
+                evidence.evidenceType.ontologySource = ontologySource_
+                evidence.evidenceType.id = id_
+
+                evidence.evidenceType.name = ''
+                if 'label' in annotation[evidenceURI]:
+                    evidence.evidenceType.name = \
+                        annotation[evidenceURI]['label']
+                fpa.evidence.append(evidence)
+                if not protocol.Evidence.validate(evidence.toJsonDict()):
+                    raise exceptions.RequestValidationFailureException(
+                           evidence.toJsonDict(), protocol.Evidence)
+
+        return fpa
+
+    def namespaceSplit(self, _url, _separator='/'):
+        """
+        given a url return the id of the resource and the ontology source
+        """
+        o = urlparse(_url)
+        _id = o.path.split(_separator)[-1]
+        _ontologySource = urlunsplit([o[0],
+                                      o[1],
+                                      o[2].replace(_id, ''),
+                                      o[4], ''])
+        return(_id, _ontologySource)
```

Comment:
No need for brackets here,
```python
return id_, ontologySource
```
works fine.

Also, lots of unnecessary underscores here. We just need to suffix id because of clash with built in function. All the other variable here should be without underscores.